### ISSUE TITLE
Running `cypress open` with `--e2e` and `-b chrome` breaks file watching

### ISSUE THREAD

**Issue by jannispl** [cypress-io/cypress#28768]
### Current behavior

Currently, running `cypress open` with `--e2e` and `-b chrome` breaks file watching. Specs are no longer automatically re-ran when the underlying spec file changes.

### Desired behavior

In this situation, Cypress should continue re-running specs when the underlying spec files change.

### Test code to reproduce

* Clone `cypress-test-tiny` repo
* run `npm i -D cypress`
* run `npx cypress open --e2e -b chrome`
* make a change to `cypress/e2e/spec.cy.js`

### Cypress Version

13.6.3

### Node version

20.9.0

### Operating System

Linux 5.15.90.1-microsoft-standard-WSL2

### Debug Logs

```shell
Too long for posting here.
```

### Other

I suspect this is caused by additional config initialization logic that is somehow skipped when providing `--e2e` and `-b chrome`. When not providing either of these arguments, Cypress will open a window that asks for testing type & browser, before showing an additional loading screen "Initializing config...".

**Comment by jennifer-shehane** [comment]
@jannispl I'm not able to reproduce this issue. Does this reliably always happen for you? I wonder if it's a timing issue where this might happen sometimes. 🤔

**Comment by jannispl** [comment]
> @jannispl I'm not able to reproduce this issue. Does this reliably always happen for you? I wonder if it's a timing issue where this might happen sometimes. 🤔

Yeah, it happens all the time. I wonder if WSL is playing a role here? I am storing my files inside the Linux partition though.

**Comment by jannispl** [comment]
I can still reproduce this issue with Cypress 13.13.1.

**Comment by jannispl** [comment]
@jennifer-shehane This issue is still labeled "stage: needs information". Could you clarify what information is needed?

**Comment by ItaloBorrelli** [comment]
Same issue when running `npx cypress open --component` and opening Chrome. Changes not detected and the spec never reruns. If I reload the test it seems webpack entirely recompiles with:
```
 [webpack-dev-middleware] wait until bundle finished: /__cypress/src/index.html
```

The same thing does not happen in Firefox or Electron. In those browsers specs are automatically reloaded, so I don't think it's a webpack issue. Happens for myself and other developers using the same Cypress version (13.13.2).
